navigating-many-to-one.mdx

```diff
@@ -0,0 +1,335 @@
+            "likes": 20,
+            "title": "Modern database workflows with GitHub, Vercel, Netlify, and Xata"
+        }
+    ]
+}
+```
+
+## The N + 1 problem
+
+What if we wanted to list the authors and their posts? You would need to query the _users_ table in order to retrieve all authors:
+
+```json
+// Request
+POST /db/blogs:main/tables/users/query
+{
+  "columns": ["*"]
+}
+
+// Response (Xata fields have been omitted for clarity)
+{
+    "records": [
+        {
+            "id": "rec_cie05krjtojbm41fv2q0",
+            "name": "Tudor Golubenco"
+        },
+        {
+            "id": "rec_cie05ljjtojbm41fv2qg",
+            "name": "Joan Edwards"
+        },
+        {
+            "id": "rec_cii1ki3jtoj0t82ae3b0",
+            "name": "Alex Francoeur"
+        }
+    ]
+}
+```
+
+And then, for each user that has been returned, make a separate request in the _posts_ table to retrieve the posts from each author, similar to:
+
+```json
+// Request
+POST /db/blogs:main/tables/posts/query
+{
+  "columns": ["*"],
+  "filter": {
+      "author": "rec_cie05krjtojbm41fv2q0"
+  }
+}
+
+// Response (Xata fields have been omitted for clarity)
+{
+    "records": [
+        {
+            "author": {
+                "id": "rec_cie05krjtojbm41fv2q0"
+            },
+            "id": "rec_cie05srjtojbm41fv2t0",
+            "likes": 13,
+            "title": "Postgres schema changes are still a PITA"
+        },
+        {
+            "author": {
+                "id": "rec_cie05krjtojbm41fv2q0"
+            },
+            "id": "rec_cie05v3jtojbm41fv2tg",
+            "likes": 15,
+            "title": "On the performance impact of REPLICA IDENTITY FULL in Postgres"
+        },
+        {
+            "author": {
+                "id": "rec_cie05krjtojbm41fv2q0"
+            },
+            "id": "rec_cii1kajjtoj0t82ae3ag",
+            "likes": 6,
+            "title": "Semantic or keyword search for finding ChatGPT context. Who searched it better?"
+        }
+    ]
+}
+```
+
+We are performing "1" (get the authors) + "N" (get the authors's posts) requests. This is inconvenient and also quite inefficient, since the number of requests depends on the number of user records.
+
+There are workarounds depending on the data set and the amount of code we are willing to write, like getting _all_ the authors and _all_ the posts, or using the `IN` operator and then processing the response in the app. But it would be better if our API would help us getting exactly what we want.
+
+# Navigate links backwards
+
+Since the relationship is already defined by the `link` column, it’s a matter of providing a good API experience to get this data more easily. We have introduced [column expressions](/docs/concepts/data-model#links-and-relations) in the query endpoint so the `columns` field is no longer restricted to a list of column names. Now we are able to provide a JSON object to better define the projection we want.
```

Comment:
Just noticed the docs only has Python and REST examples. If there's TS SDK support, we should update too.